src/offset/local/unix.rs

```diff
@@ -126,12 +126,13 @@ impl Cache {
         }
 
         if !local {
-            let offset = FixedOffset::east(
+            let offset = FixedOffset::east_opt(
                 self.zone
                     .find_local_time_type(d.timestamp())
                     .expect("unable to select local time type")
                     .offset(),
-            );
+            )
+            .unwrap();
```

Comment:
Interesting question - I don't think `LocalResult::None` is ideal here as it is meant to represent the specific situation where the local timestamp is invalid due to the local offset moving east, however it may be better than panicking, and so could be reasonable here, and then later we could allow more kinds of errors from this function to better signify what has gone wrong